### PR TITLE
Add otel tracing via file exports to nodejs-corepack

### PR DESCRIPTION
**Pull request by joshwlewis**
This adds an `initTracer` function for Node.js buildpacks to use, which sets up an OpenTelemetry file exporter for tracing data. 

Additionally, this adds some tracing to the`heroku/nodejs-corepack` buildpack.

### CODE REVIEW

File: buildpacks/nodejs-corepack/src/main.rs [hunk 1]
```diff
@@ -37,60 +40,76 @@ impl Buildpack for CorepackBuildpack {
...
-                                .provides(pkg_mgr)
-                                .build(),
-                        )
-                        .build()
-                },
-            )
-        } else {
-            DetectResultBuilder::fail().build()
-        }
+        let tracer = init_tracer(format!("{}", context.buildpack_descriptor.buildpack.id));
```
Comment (colincasey) [inline buildpacks/nodejs-corepack/src/main.rs:43]: this is where i think tighter integration with `libcnb.rs` would be helpful. auto-initializing the tracer (or disabling it if opt-in/out can be configured) and pre-populating a root span with the buildpack id (and maybe some other useful default attributes).

File: buildpacks/nodejs-corepack/src/main.rs [hunk 2]
```diff
@@ -37,60 +40,76 @@ impl Buildpack for CorepackBuildpack {
...
-            layer_name!("mgr"),
-            ManagerLayer {
-                package_manager: pkg_mgr,
-            },
-        )?;
-        let mgr_env = mgr_layer.env.apply(Scope::Build, env);
-        cmd::corepack_prepare(&mgr_env).map_err(CorepackBuildpackError::CorepackPrepare)?;
-
-        BuildResultBuilder::new().build()
+        let tracer = init_tracer(format!("{}", context.buildpack_descriptor.buildpack.id));
```
Comment (colincasey) [inline buildpacks/nodejs-corepack/src/main.rs:70]: will this clobber the tracer created during `detect`? or does that file get consumed before this happens?

**Review by colincasey — COMMENTED** [review]
seems fairly straightforward.  

also interesting is the `opentelemetry` crate allows you to [get a reference to the active span from anywhere](https://docs.rs/opentelemetry/latest/opentelemetry/trace/fn.get_active_span.html).

File: buildpacks/nodejs-corepack/src/main.rs (hunk 2 above)
Comment (joshwlewis) [reply buildpacks/nodejs-corepack/src/main.rs:70]: `detect` and `build` are different binaries. They shouldn't be in scope at the same time.

File: buildpacks/nodejs-corepack/src/main.rs (hunk 1 above)
Comment (joshwlewis) [reply buildpacks/nodejs-corepack/src/main.rs:43]: Yeah, totally agree! I want to trial it here before bringing it into libcnb.rs for now though.

**Review by colincasey — APPROVED** [review]

**Pull request merged**